test/v3/createOrAmendCodingOut/def1/CreateOrAmendCodingOutValidatorFactorySpec.scala

```diff
@@ -0,0 +1,306 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package v3.createOrAmendCodingOut.def1
+
+import api.models.domain.{Nino, TaxYear}
+import api.models.errors._
+import api.models.utils.JsonErrorValidators
+import config.MockAppConfig
+import play.api.libs.json.{JsObject, JsPath, JsValue, Json}
+import support.UnitSpec
+import v3.createOrAmendCodingOut.CreateOrAmendCodingOutValidatorFactory
+import v3.createOrAmendCodingOut.def1.model.request.{Def1_CreateOrAmendCodingOutRequestBody, Def1_CreateOrAmendCodingOutRequestData, TaxCodeComponents}
+
+class CreateOrAmendCodingOutValidatorFactorySpec extends UnitSpec with JsonErrorValidators with MockAppConfig {
```

Comment:
Please change this to Def1_CreateOrAmendCodingOutValidatorSpec and also add CreateOrAmendCodingOutValidatorFactorySpec (use bsas api as a reference). 
Also, where is CreateOrAmendCodingOutConnectorSpec and CreateOrAmendCodingOutServiceSpec?
Please add the RequestBodySpec too.

Please ensure the other endpoints all have a separate Def1_ValidatorSpec and ValidatorFactorySpec, request and response specs, and other required specs (most of them have one or the other missing!)